src/jit/codegenxarch.cpp

```diff
@@ -7393,11 +7427,25 @@ CodeGen::genCkfinite(GenTreePtr treeNode)
     // If exponent is all 1's, throw ArithmeticException
     genJumpToThrowHlpBlk(EJ_je, SCK_ARITH_EXCPN);
 
+#ifdef _TARGET_AMD64_
     // if it is a finite value copy it to targetReg
     if (treeNode->gtRegNum != op1->gtRegNum)
     {
         inst_RV_RV(ins_Copy(targetType), treeNode->gtRegNum, op1->gtRegNum, targetType);
     }
+#else // !_TARGET_AMD64_
+    if (targetType == TYP_DOUBLE)
+    {
+        inst_RV_RV_IV(INS_shufps, EA_16BYTE, targetReg, targetReg, 0xb1);
```

Comment:
We don't always have to generate shufps here.  If targetReg != op1->gtRegNum, we can generate "movaps targetReg, op1->gtRegnum".  Note that movaps is faster and smaller than shufps.